File: src/main/java/ControleGastos/ControleGastos/Conta/Conta.java
package ControleGastos.ControleGastos.Conta;

import ControleGastos.ControleGastos.Transacao.Transacao;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Data
public class Conta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nome;
    private String sobrenome;
    private String telefone;
    private String senha;
    private String email;
    private BigDecimal total;
    @OneToMany(mappedBy = "conta", cascade = CascadeType.ALL)
    private List<Transacao> transacoes;
}

File: frontend-controle-gastos/src/app/components/Conta/conta.ts
import { Transacao } from "../Transacoes/transacao"

export interface Conta{
  id?: number,
  nome: string,
  sobrenome:string,
  telefone: string,
  email: string,
  senha: string,
  total?: number,
  transacoes?: Transacao[]
}
